### PR TITLE
Fix #898 - add mt_bam to scout config

### LINKED ISSUE #898: Bam files missing for mtDNA
I have noticed that BAM files for mtDNA are missing for several cases. I have now checked and all cases uploaded in Scout after 20th of November miss BAM files for mtDNA. Could you please check what has happened.

### PR DESCRIPTION
**Pull request by dnil**
This PR adds/fixes issue #898 - missing mt_bam fields in scout config

### How to prepare for test
- [x] ssh to hasta (depending on type of change)
- [x] install on stage:
`bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-cg-stage.sh [THIS-BRANCH-NAME]`

### How to test
- [x] login to hasta or clinical-db
- [x] activate stage
- [x] cg upload scout deepbeagle or other common test case with mt_bam files

### Expected test outcome
- [x] ensure that the freshly made scout load config file contains mt_bam
- [x] ensure that the test case shows a bam file for a mitochondrial variant in scout-stage
- [x] Take a screenshot and attach or copy/paste the output.

## Review
- [ ] code approved by
- [x] tests executed by DN
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a
- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [x] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan
- [ ] Document in ...
- [ ] Deploy this branch
- [ ] Inform to ...

### CODE REVIEW

**Comment by dnil** [comment]
I'll try to add a test as well..

**Comment by dnil** [comment]
No precommit hook or auto-black on this one? I'm surprised: will fix black as well! 😸

**Comment by dnil** [comment]
Present issue notwithstanding, really good job with the remake of the cg-scout interaction @moonso and @Mropat!

This is ready for review!

**Comment by dnil** [comment]
Tested on hasta stage:

Before patch:
```
/home/proj/stage/rare-disease/cases/deepbeagle
$ cat scout_load.yaml.orig
...
samples:
- sample_id: ACC5341A12
  father: ACC5557A2
  mother: ACC5557A1
  sample_name: 2019-03867-02
  sex: female
  phenotype: affected
  alignment_path: /home/proj/stage/housekeeper-bundles/deepbeagle/2020-12-03/ACC5341A12_lanes_1234_sorted_md_brecal.cram
  chromograph: /home/proj/stage/housekeeper-bundles/deepbeagle/2020-12-03/ACC5341A12_lanes_1234_sorted_md_brecal_tcov_1.png
  vcf2cytosure: /home/proj/stage/housekeeper-bundles/deepbeagle/2020-12-03/deepbeagle_cyto.ACC5341A12.cgh
  analysis_type: wgs
  tissue_type: blood
...
```
Appearance in scout-stage:
Screenshot 2020-12-17 at 11 35 36

Apply patch, re-upload a case with mt_bam:
```
$ cg upload scout --re-upload deepbeagle
...
Generating madeline input lines
Running command /home/proj/stage/servers/resources/hasta.scilifelab.se/madeline2 --color --nolabeltruncation --outputprefix /home/daniel.nilsson/tmp/tmpqvgwn3lu/madeline /home/daniel.nilsson/tmp/tmp7tqtc_z0
Scout load config /home/proj/stage/rare-disease/cases/deepbeagle/scout_load.yaml already exists
Deleting old load config
Save Scout load config to /home/proj/stage/rare-disease/cases/deepbeagle/scout_load.yaml
Upload file to housekeeper: /home/proj/stage/rare-disease/cases/deepbeagle/scout_load.yaml
Adding load config to housekeeper
Fetching files from bundle deepbeagle
Fetching files with tags in [scout-load-config]
Fetching files from version 27925
Found config file: <File(id=525219, path='deepbeagle/2020-12-03/scout_load.yaml', checksum=None, to_archive=False, version_id=27925)>
Fetching file 525219
Deleting file /home/proj/stage/housekeeper-bundles/deepbeagle/2020-12-03/scout_load.yaml form disc
Deleting file 525219 from housekeeper
Created new bundle version dir: /home/proj/stage/housekeeper-bundles/deepbeagle/2020-12-03
Linked file: /home/proj/stage/rare-disease/cases/deepbeagle/scout_load.yaml -> /home/proj/stage/housekeeper-bundles/deepbeagle/2020-12-03/scout_load.yaml
Added scout load config to housekeeper: /home/proj/stage/rare-disease/cases/deepbeagle/scout_load.yaml
----------------- UPLOAD -----------------------
Fetching files from bundle deepbeagle
Fetching files with tags in [scout-load-config]
Fetching files from version 27925
uploading case deepbeagle to scout
Running command /home/proj/bin/conda/envs/S_scout/bin/scout --config /home/proj/stage/servers/config/hasta.scilifelab.se/scout-stage.yaml export cases --json --case-id deepbeagle
Validating case deepbeagle
update existing Scout case
Running command /home/proj/bin/conda/envs/S_scout/bin/scout --config /home/proj/stage/servers/config/hasta.scilifelab.se/scout-stage.yaml load case /home/proj/stage/housekeeper-bundles/deepbeagle/2020-12-03/scout_load.yaml --update
```

After patch:
```
$ cat scout_load.yaml
- sample_id: ACC5341A12
  father: ACC5557A2
  mother: ACC5557A1
  sample_name: 2019-03867-02
  sex: female
  phenotype: affected
  alignment_path: /home/proj/stage/housekeeper-bundles/deepbeagle/2020-12-03/ACC5341A12_lanes_1234_sorted_md_brecal.cram
  mt_bam: /home/proj/stage/housekeeper-bundles/deepbeagle/2020-12-03/ACC5341A12_lanes_1234_sorted_md_brecal_subsample_MT.bam
  chromograph: /home/proj/stage/housekeeper-bundles/deepbeagle/2020-12-03/ACC5341A12_lanes_1234_sorted_md_brecal_tcov_1.png
  vcf2cytosure: /home/proj/stage/housekeeper-bundles/deepbeagle/2020-12-03/deepbeagle_cyto.ACC5341A12.cgh
  analysis_type: wgs
  tissue_type: blood
...
```
And looks good on scout stage:
Screenshot 2020-12-17 at 11 35 36

File: CHANGELOG.md [hunk 1]
```diff
@@ -13,7 +13,7 @@ Try to use the following format:
 ### Added
 
 ### Fixed
-
+- Fixed exporting reduced mt_bam to scout config
```
Comment (moonso) [inline CHANGELOG.md:16]: We try to preserve these lines so please make a soup and insert the changes below

File: tests/apps/scout/test_scout_load_config.py [hunk 2]
```diff
@@ -3,24 +3,25 @@
...
+    sample = sample_dict
 
     # WHEN validating the sample data
     ind_obj = scout_load_config.ScoutIndividual(**sample)
 
     # THEN assert that the display name is correct
     assert ind_obj.sample_name == sample["sample_name"]
+
+
+def test_validate_mt_bam(sample_dict):
```
Comment (moonso) [inline tests/apps/scout/test_scout_load_config.py:18]: 👍

**Review by moonso — APPROVED** [review]

**Comment by Mropat** [comment]
Deployed on prod
image

**Pull request merged**